src/state/service.rs

```diff
@@ -94,18 +101,18 @@ pub struct ServiceStore {
     /// Allows for lookup of services by network address, the service's xds secondary key.
     /// this map stores the same services as `services_by_hostname` so we can mutate the services in both maps
     /// at once when a service is updated or workload updates affect a service's endpoints.
-    by_vip: HashMap<NetworkAddress, Arc<RwLock<Service>>>,
+    by_vip: HashMap<NetworkAddress, Arc<Service>>,
 
     /// Allows for lookup of services by namespaced and hostname, the service's xds primary key.
     /// this map stores the same services as `services_by_ip` so we can mutate the services in both maps
     /// at once when a service is updated or workload updates affect a service's endpoints.
-    by_host: HashMap<String, Vec<Arc<RwLock<Service>>>>,
+    by_host: HashMap<String, Vec<Arc<Service>>>,
```

Comment:
if service is immutable now, do we even need the `Arc` anymore? why can't we just put this on the stack?

Reply:
I think it's still useful to reduce the overall storage required.

Reply:
I think that's reasonable :+1: